src/objprint/objprint.py

```diff
@@ -130,12 +132,16 @@ def objstr(self, obj: Any, **kwargs) -> str:
         return self._objstr(obj, memo, indent_level=0, cfg=cfg)
 
     def _objstr(self, obj: Any, memo: Optional[Set[int]], indent_level: int, cfg: _PrintConfig) -> str:
+        # If a custom formatter is registered for the object's type, use it directly
+        if cfg.inherit and self.type_formatter:
+            obj_type = type(obj)
+            for cls in obj_type.__mro__:
+                if cls in self.type_formatter:
+                    return self.type_formatter[cls](obj)
         # If it's builtin type, return it directly
         if isinstance(obj, str):
             return f"'{obj}'"
-        elif isinstance(obj, int) or \
-                isinstance(obj, float) or \
-                obj is None:
+        elif isinstance(obj, (int, float)) or obj is None:
```

Comment:
This is the correct way to check types, but it's note related to this PR. To avoid confusion, we should probably submit this change in another PR(and maybe fix all occurences this kind).

Reply:
Acknowledged.